#169523245 user is able to view single entry

Ntare22 opened the pull request:
> #### What does this PR do?
* Allows user to view a single entry.
> #### Description of Task to be completed
* add query to view a single entry
* receive entryId from params and turn it into integer
* retrieve user entries
> #### How should this be manually tested?
* clone this repo
* create .env file
* set your pwdSalt, secrectKey and dbURL (postgres credentials link) in the .env file
* enter "npm run dev" in your commnad line.
( open postman and create a user by sending json file on this link "localhost:3000/api/auth/signup"
* in postman, create an entry on this link "localhost:3000/api/v2/entries"
* create a user by sending json file and select POST
* view single entry on this link "localhost:3000/api/v2/entries/1"
* choose GET
* click send
> #### Any background context you want to provide?
* N/A
> #### What are the relevant pivotal tracker stories?
[#169523245](https://www.pivotaltracker.com/story/show/169523245)
> #### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/30793016/68209020-2e3a3c80-ffdb-11e9-9345-78cc2ec72b5b.png)

Commit:
feat(user is able to view single entry):

add query to view a single entry
recieve entryId from params and turn it into integer
retrieve user entries

[ Finishes #169523245 ]

mcaleb808 commented on server/controllers/entryController.js:
```diff
@@ -139,8 +139,33 @@ class EntryController {
+
+            const getEntriesQuery = `SELECT * FROM mydiaryentries WHERE entryid=$1 AND userid=$2;`;
+            const getUserEntries = await pool.query(getEntriesQuery, values);
+            // return console.log(getUserEntries);
```
@Ntare22 Please avoid pushing commented code

mcaleb808 commented on server/controllers/entryController.js:
```diff
@@ -139,8 +139,33 @@ class EntryController {
+                error: error.message
+            });
+        }
+
```
@Ntare22 add a scenario where the entry doesn't exist

Commit:
feat(view single entry):

add if statement to check if the entry is available

[ Finishes #169523245 ]

mcaleb808 commented on server/controllers/entryController.js:
```diff
@@ -139,8 +139,40 @@ class EntryController {
+
+            const getEntriesQuery = `SELECT * FROM mydiaryentries WHERE entryid=$1 AND userid=$2;`;
+            const getUserEntries = await pool.query(getEntriesQuery, values);
+            // return console.log(getUserEntries.rows[0].entryid);
```
@Ntare22 we still have commented code

mcaleb808 commented on server/controllers/entryController.js:
```diff
@@ -139,8 +139,40 @@ class EntryController {
+            // return console.log(getUserEntries.rows[0].entryid);
+
+            if (!getUserEntries.rows[0]) {
+                return res.status(400).send({
```
@Ntare22 let replace `400` to `404` status code as we are dealing with resource not found instead of a bad request.

Commit:
feat(view single entry):

added some tests
fix a bug with the signup
added an error for when entry doesn't exist

[ Finishes #169523245 ]
